src/connector/with_options_source.yaml

```diff
@@ -537,6 +537,7 @@ NatsProperties:
   - name: consumer.filter_subjects
     field_type: Vec<String>
     required: false
+    default: Default::default
```

Comment:
Does this PR imply the following problems:
* Without the `serde(default)` the `required: false` here is actually misleading
* Other attributes without `serde(default)` have the same problem

Reply:
Based on our testing of this, the other attributes are unaffected. It seems that only the vector types are affected due - my guess is that this is somehow needed due to the custom deserializer.

I drew reference from the Iceberg sink that uses a similar pattern https://github.com/risingwavelabs/risingwave/blob/63b4d47ea9f54f567983e097e15d8dc9b0384849/src/connector/src/sink/iceberg/mod.rs#L129-L141

Reply:
Thanks! Yes, it seems only required when using `deserialize_with`. We will see if we can fix the `required: false` in yaml.

Reply:
Anyway just to confirm, these attributes are optional so I think it's fine to keep `required: false` :)

Thanks for taking a look!

Reply:
Yes these 2 are now truly optional with `default`. I meant fixing the yaml generator when `default` is missing but `deserialize_with` is present.